python/vyos/ifconfig/control.py

```diff
@@ -49,6 +49,18 @@ def _popen(self, command):
         return popen(command, self.debug)
 
     def _cmd(self, command):
+        import re
+        if 'netns' in self.config:
+            # This command must be executed from default netns 'ip link set dev X netns X'
```

Comment:
IMHO this is too much logic in two places.

looking at e.g.

```python
        elif is_intf_addr_assigned(self.ifname, addr, netns=self.config.get('netns')):
            exec_within_netns = f'ip netns exec {self.config.get("netns")}' if self.config.get('netns') else ''
            self._cmd(f'{exec_within_netns} ip addr del "{addr}" dev "{self.ifname}"')
```

you specify the netns command outside. In this helper you check if it was specified and execute it or add it if it needs to be added. Why not keep the existing code "as is" in the above case and only rely on you new logic here to "add" the netns command if required?

Thinking about execution time you always to regular matching on "every" command that traverses `self._cmd()` instead of simply extending a string. Can you please ensure that `ip netns exec` is only prepended once in this function and all other functions stay as they were?

That was one of the reasons to design a command wrapper so only single changes would be required when mangling the OS commands.

Reply:
It doesn't work otherwise.
There are methods in the `interface.py` the method `ifconfig.interface.get_foo` will work fine with netns command which added a namespace from `control._cmd`
But if this method is used for example in another method, for example `set_foo` will be executed in netns (control._cmd), but if he called `get_foo` or external `ifconfig.validate.xxx` that method will be executed in the default namespace.

`get_foo` netns ok `control._cmd`
`set_foo` uses `get_foo` - **set_foo** will be executed in expected netns, the **get_foo** as I understand will be executed in default netns

`is_intf_addr_assigned` will be executed without nents even if we have `control._cmd`  the set the correct netns.
Yes, it is annoying. It looks ugly, maybe external lib [pyroute2](https://github.com/svinota/pyroute2) can help us with all netns magic